app/src/debug/java/timur/gilfanov/messenger/debug/DebugDataRepositoryImpl.kt

```diff
@@ -0,0 +1,278 @@
+package timur.gilfanov.messenger.debug
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Singleton
+import kotlin.random.Random
+import kotlin.time.Clock
+import kotlinx.collections.immutable.toPersistentList
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import timur.gilfanov.messenger.data.source.local.LocalDataSources
+import timur.gilfanov.messenger.data.source.local.LocalDebugDataSource
+import timur.gilfanov.messenger.data.source.remote.RemoteDebugDataSource
+import timur.gilfanov.messenger.debug.datastore.DebugPreferences
+import timur.gilfanov.messenger.domain.entity.ResultWithError
+import timur.gilfanov.messenger.domain.entity.chat.Chat
+import timur.gilfanov.messenger.util.Logger
+
+@Suppress("TooManyFunctions", "LongParameterList", "TooGenericExceptionCaught") // Debug code
```

Comment:
`TooGenericExceptionCaught` can be removed after use specific exceptions to catch system database exceptions. 
```suggestion
@Suppress("TooManyFunctions", "LongParameterList") // Debug code
```